### DISCUSSION TITLE
input component event handler

### DISCUSSION THREAD

**Discussion by novetto** [reflex-dev/reflex#1719] [Q&A] [answered]
From the [input](https://reflex.dev/docs/library/forms/input/#input) example:

```
class InputState(rx.State):
    text: str = "Type something..."


def index():
    return rx.vstack(
        rx.text(InputState.text, color_scheme="green"),
        rx.input(
            value=InputState.text,
            on_change=InputState.set_text,
        ),
    )
```

Where the `set_text` attribute is coming from?

**Comment by masenf** [comment, marked as answer]
`set_text` is implicit; every var in the state gets a corresponding `set_X` method that can be used to set it.

**Reply by novetto** [reply]
Thanks. I think it is better to add this info in the documentation.

**Reply by masenf** [reply]
It's in the docs: https://reflex.dev/docs/state/events/#setters

**Comment by novetto** [comment]
How about this, I created a method `change_text` as a way to change the `text` value.  This method has a parameter `val` to be received by `text` as a new value. Why this setup worked even though the `on_change` is set to `change_text` without argument?.

I like explicit definition of methods and parameters.

```
class InputState(rx.State):
    text: str = "Type something..."

    def change_text(self, val):
        self.text = val


def index():
    return rx.vstack(
        rx.text(InputState.text, color_scheme="green"),
        rx.input(
            value=InputState.text,
            on_change=InputState.change_text
        ),
    )
```
